influence_benchmark/stats/preferences_per_iteration.py

```diff
@@ -8,133 +8,141 @@
+def process_iteration_data(trajectory_path: Path, top_n: int) -> Optional[Tuple[int, float, float, float, float]]:
     """Process data for a single iteration.
     Returns
-        overall_expected_pref: reward values averaged over all trajectories
-        top_n_avg: reward value averaged over the top n trajectories
         n_trajectories: number of trajectories in the iteration
+        rew_avg_all_trajs: reward values averaged over all trajectories
+        rew_avg_top_trajs: reward value averaged over the top n trajectories
+        infl_avg_all_trajs: influence score values averaged over all trajectories
+        infl_avg_top_trajs: influence score value averaged over the top n trajectories
+
     """
     # Check if there are any trajectories
     if next(trajectory_path.iterdir(), None) is None:
         return None
-    # Load all trajectories from files
-    top_n_trajectories = get_top_n_trajectories(trajectory_path, top_n)
 
-    overall_expected_pref = sum(traj["reward_avg_all_trajectories"] for traj in top_n_trajectories) / len(
-        top_n_trajectories
-    )
-    top_n_avg = sum(traj["reward_avg_selected_trajectories"] for traj in top_n_trajectories) / len(top_n_trajectories)
-    n_trajectories = sum(traj["n_trajectories"] for traj in top_n_trajectories)
+    top_n_trajs, _ = get_best_worst_n_trajectories(trajectory_path, top_n)
+    # We have averages for each initial state configuration (from the above function), and now we want to average across them
+    rew_avg_all_trajs = sum(traj["avg_rew_across_trajs_with_init_s"] for traj in top_n_trajs) / len(top_n_trajs)
+    rew_avg_top_trajs = sum(traj["avg_rew_across_top_trajs_with_init_s"] for traj in top_n_trajs) / len(top_n_trajs)
 
-    return (
-        overall_expected_pref,
-        top_n_avg,
-        n_trajectories,
-    )
+    infl_avg_all_trajs = sum(traj["avg_infl_across_trajs_with_init_s"] for traj in top_n_trajs) / len(top_n_trajs)
+    infl_avg_top_trajs = sum(traj["avg_infl_across_top_trajs_with_init_s"] for traj in top_n_trajs) / len(top_n_trajs)
+
+    n_trajs = sum(traj["n_trajectories"] for traj in top_n_trajs)
 
+    return n_trajs, rew_avg_all_trajs, rew_avg_top_trajs, infl_avg_all_trajs, infl_avg_top_trajs
 
-def analyze_run(run_name: str, top_n: int = 1, print_out=True) -> Tuple[List[int], List[float], List[float]]:
+
+def analyze_run(
+    run_name: str, top_n: int = 1, print_out=True
+) -> Tuple[List[int], List[float], List[float], List[float], List[float]]:
     """Analyze a complete run and return iteration data."""
     data_path = PROJECT_DATA / "trajectories" / run_name
     iterations = sorted(int(d.name) for d in data_path.iterdir() if d.is_dir() and d.name.isdigit())
 
-    expected_prefs = []
-    top_n_averages = []
+    all_rew_avg_all_trajs = []
+    all_rew_avg_top_trajs = []
+    all_infl_avg_all_trajs = []
+    all_infl_avg_top_trajs = []
```

Comment:
Same here, all of this could be a dictionary of lists, created with `defaultdict(list)`

Reply:
Agreed. To be left for further work.